Kerberos.NET/Client/Krb5CredentialCache.cs

```diff
@@ -0,0 +1,600 @@
+                {
+                    sessionKey = entryValue.SessionKey;
+
+                    if (!this.Credentials.Any())
+                    {
+                        this.DefaultPrincipalName = FromResponse(asRep, asRep.CName);
+                    }
+                }
+                else
+                {
+                    var encKdcRepPart = entryValue.KdcResponse.EncPart.Decrypt(
+                        entryValue.SessionKey.AsKey(),
+                        entryValue.SessionKey.Usage,
+                        d => KrbEncTgsRepPart.DecodeApplication(d)
+                    );
+
+                    sessionKey = encKdcRepPart.Key;
+                }
+
+                this.Credentials.Add(new Krb5Credential
+                {
+                    Ticket = entryValue.KdcResponse.Ticket.EncodeApplication(),
+                    KeyBlock = new KeyValuePair<EncryptionType, ReadOnlyMemory<byte>>(sessionKey.EType, sessionKey.KeyValue),
+                    Client = FromResponse(entryValue.KdcResponse, entryValue.KdcResponse.CName),
+                    Server = FromResponse(entryValue.KdcResponse, entryValue.KdcResponse.Ticket.SName),
+                    AuthData = new List<KrbAuthorizationData>(),
+                    EndTime = entry.Expires,
+                    RenewTill = entry.RenewUntil ?? DateTimeOffset.MinValue,
+                    Addresses = new List<KrbHostAddress>(),
+                    SecondTicket = Array.Empty<byte>()
+                });
+            }
+        }
+
+        private static PrincipalName FromResponse(KrbKdcRep asRep, KrbPrincipalName name)
+        {
+            return new PrincipalName(asRep.CName.Type, asRep.CRealm, name.Name);
+        }
+
+        internal object GetCacheItem(string key)
+        {
+            Krb5Credential cred = FindCredential(key);
+
+            if (cred == null)
+            {
+                return cred;
+            }
+
+            return new KerberosClientCacheEntry
+            {
+                KdcResponse = new KrbTgsRep
+                {
+                    Ticket = KrbTicket.DecodeApplication(cred.Ticket),
+                    CName = KrbPrincipalName.FromString(cred.Client.FullyQualifiedName),
+                    CRealm = cred.Client.Realm,
+                    EncPart = new KrbEncryptedData { }
+                },
+                SessionKey = new KrbEncryptionKey
+                {
+                    EType = cred.KeyBlock.Key,
+                    KeyValue = cred.KeyBlock.Value
+                },
+                SName = KrbPrincipalName.FromString(cred.Server.FullyQualifiedName)
+            };
+        }
+
+        private Krb5Credential FindCredential(string key)
+        {
+            return this.Credentials.FirstOrDefault(c => c.Server.FullyQualifiedName == key);
+        }
+
+        internal bool Contains(TicketCacheEntry entry)
+        {
+            Krb5Credential cred = FindCredential(entry.Key);
+
+            return cred != null;
```

Comment:
```suggestion
            return !(cred is null);
```
With C# 9 one could write `cred is not null`.

You can keep the code as is, I just want to point the "new possibilities" out.